phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -1339,4 +1362,193 @@ public Expression visit(SubqueryParseNode node) throws SQLException {
+        Expression lhs = children.get(0);
+        if (rhsNode instanceof BindParseNode) {
+              context.getBindManager().addParamMetaData((BindParseNode)rhsNode, lhs);
+        }
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(!(PVarchar.INSTANCE.isComparableTo(children.get(1).getDataType()))){
+     		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+      	}
+        Expression expression = new JsonSingleKeySearchExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
+    @Override
+    public Expression visitLeave(JsonContainWithinRightParseNode node, List<Expression> children) throws SQLException {
+    	ParseNode lhsNode = node.getChildren().get(0);
+    	ParseNode rhsNode = node.getChildren().get(1);
+    	Expression lhs = children.get(0);
+    	Expression rhs = children.get(1);
+    	if (lhsNode instanceof BindParseNode) {
+    		context.getBindManager().addParamMetaData((BindParseNode)lhsNode, rhs);
+    	}
+    	if (rhsNode instanceof BindParseNode) {
+    		context.getBindManager().addParamMetaData((BindParseNode)rhsNode, lhs);
+    	}
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(1).getDataType()))){
+      		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+        Expression expression = new JsonContainWithinRightExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
+    @Override
+    public Expression visitLeave(JsonContainWithinLeftParseNode node, List<Expression> children) throws SQLException {
+    	ParseNode lhsNode = node.getChildren().get(0);
+    	ParseNode rhsNode = node.getChildren().get(1);
+    	Expression lhs = children.get(0);
+    	Expression rhs = children.get(1);
+    	if (lhsNode instanceof BindParseNode) {
+    		context.getBindManager().addParamMetaData((BindParseNode)lhsNode, rhs);
+    	}
+    	if (rhsNode instanceof BindParseNode) {
+    		context.getBindManager().addParamMetaData((BindParseNode)rhsNode, lhs);
+    	}
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(0).getDataType()))){
+    		throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+       	}
+    	if(!(PJson.INSTANCE.isComparableTo(children.get(1).getDataType()))){
+   		 throw TypeMismatchException.newException(children.get(0).getDataType(), children.get(1).getDataType());
+    	}
+        Expression expression = new JsonContainWithinLeftExpression(children);
+        return wrapGroupByExpression(expression);
+    }
+    
```

Comment:
For JsonMultiKeySearchOrParseNode  and JsonMultiKeySearchAndParseNode do you need to check if the lhsNodeis / rhsNode are instances of BindParseNode ?